main/src/library/Pkger/SemVer.flix

```diff
@@ -0,0 +1,92 @@
+namespace Pkger/SemVer {
+    ///
+    /// Represents an instance of a Semantic Versioning (SemVer) version.
+    /// It is currently missing pre-release identifiers and build metadata
+    ///
+    pub enum SemVer {
+        case SemVer(Int, Int, Int)
+    }
+
+
+    ///
+    /// Compare two SemVer instances according to the ordering rules in the specification
+    ///
+    def __cmp(l: SemVer, r: SemVer): Int32 = match (l, r) {
+        case (SemVer(lMajor, lMinor, lPatch), SemVer(rMajor, rMinor, rPatch)) => {
+            let major = lMajor <=> rMajor;
+            if (major == 0) {
+                let minor = lMinor <=> rMinor;
+                if (minor == 0) {
+                    lPatch <=> rPatch
+                } else {
+                    minor
+                }
+            } else {
+                major
+            }
+        }
+    }
+
+    ///
+    /// Check whether the `specific` SemVer is compatible with the `bound` SemVer
```

Comment:
What does it mean to be compatible?

Reply:
According to the SemVer specification. Added that to the documentation.